src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -96,5 +116,31 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
 
         return builder;
     }
+
+    private static bool IsAutoInstrumentationStartupAssemblyConfigured()
+    {
+        // If autoinstrumentation's HostingStartupAssembly is configured, assume integration was already called.
+        _autoInstrumentationStartupAssemblyConfigured ??= IsAutoInstrumentationStartupAssemblySet();
+        return _autoInstrumentationStartupAssemblyConfigured.Value;
+    }
+
+    private static bool IsNet6()
+    {
+        var frameworkDescription = FrameworkDescription.Instance;
+        return frameworkDescription.Name == ".NET" && frameworkDescription.ProductVersion.StartsWith("6");
+    }
+
+    private static bool IsHostServiceCollection(IServiceCollection builderServices)
+    {
+        _applicationLifetimeType ??= Type.GetType("Microsoft.Extensions.Hosting.Internal.ApplicationLifetime, Microsoft.Extensions.Hosting");
```

Comment:
This will load the mentioned referenced assembly, it should be fine, but, calling attention to this fact.

Reply:
Good point. I initially assumed this is fine, but started to have some doubts. I plan to improve it.

Reply:
Adjusted in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003/commits/7c21fba9d835536f6d191cc9121b48933fe8dd92